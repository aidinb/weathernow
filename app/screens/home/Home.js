import React, {useEffect} from 'react';
import {Dimensions, FlatList, View} from 'react-native';

import {observer} from 'mobx-react';
import {useTranslation} from 'react-i18next';
import {useStores} from '../../store';
import City from '../../components/City';

const {width} = Dimensions.get('window');

const Home = () => {
  const {t} = useTranslation();
  const {homeStore} = useStores();

  useEffect(() => {
    homeStore.initialLoad();
  }, []);

  return (
    <View testID={'home'}>
      <FlatList
        data={homeStore.cities}
        keyExtractor={item => item.name}
        style={{
          width,
        }}
        contentContainerStyle={{
          alignItems: 'center',
          paddingBottom: 100,
        }}
        onRefresh={async () => {
          await homeStore.getCities();
        }}
        refreshing={homeStore.loading}
        renderItem={({item}) => (
          <City
            onPress={async () => {
              await homeStore.navigateToCityDetail(item.name);
            }}
            city={item}
          />
        )}
      />
    </View>
  );
};

export default observer(Home);
