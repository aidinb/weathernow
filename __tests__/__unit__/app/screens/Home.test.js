import {
    View, Text, TextInput, BackHandler,
    StyleSheet, TouchableOpacity, Alert, Image, ImageBackground, FlatList,
} from "react-native";

import Home from '../../../../app/screens/home/Home'
// import homeStore from '../../../../app/screens/home/homeStore'
import {Provider} from 'mobx-react'


describe('Home', () => {
    let wrapper
    let HomeJson
    let HomeScreen

    beforeEach(() => {

        // creates the accessible object of all the screen element by Enzyme
        wrapper = shallow(<Provider><Home/></Provider>)

        // creates the snapshot
        HomeJson = renderer.create(<Provider><Home/></Provider>).toJSON()

        // creates the accessible object of all the screen element by react-test-renderer
        HomeScreen = renderer.create(<Provider><Home/></Provider>)
    })


    it('renders correctly SnapShot works', () => {
        expect(HomeJson).toMatchSnapshot()
    })

    it('should render', () => {
        expect(wrapper).toBeDefined()
    })


    it('should render child Element', () => {
        expect(wrapper.containsMatchingElement([
                <View/>,
                <FlatList/>
            ])
        )
    })

    it('should have View', () => {

        const view = HomeScreen.root.findAllByType(View)

        expect(view).toHaveLength(2)
    })

    it('should have FlatList', () => {

        const flatLists = HomeScreen.root.findAllByType(FlatList)

        expect(flatLists).toHaveLength(1)
    })
})
