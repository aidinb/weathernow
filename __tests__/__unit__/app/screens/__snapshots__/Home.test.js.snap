// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly SnapShot works 1`] = `
<View>
  <RCTScrollView
    contentContainerStyle={
      {
        "alignItems": "center",
        "paddingBottom": 100,
      }
    }
    data={[]}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onRefresh={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    refreshControl={
      <RefreshControlMock
        onRefresh={[Function]}
        refreshing={false}
      />
    }
    refreshing={false}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={[]}
    style={
      {
        "width": 750,
      }
    }
    viewabilityConfigCallbackPairs={[]}
  >
    <RCTRefreshControl />
    <View />
  </RCTScrollView>
</View>
`;
